src/substrait-to-d3.js

```diff
@@ -0,0 +1,296 @@
+'use strict';
+
+const { SubstraitParser } = require('./substrait-parser');
+const d3 = require('d3');
+
+// Populating map with nodes information
+function populateNodes(plan, nodes) {
+    nodes.set(plan.id, plan);
+    for (let i = 0; i < plan.inputs.length; ++i) {
+        populateNodes(plan.inputs[i], nodes);
+    }
+}
+
+// Building graph from Substrait Plan
+function buildGraph(plan) {
+    let nodes = new Map();
+    populateNodes(plan.inputs[0], nodes);
+    let edges = [];
+    nodes.forEach((value) => {
+        for (let i = 0; i < value.inputs.length; ++i) {
+            edges.push({ 'source': value.inputs[i].id, 'target': value.id })
+        }
+    })
+    return {
+        'nodes': nodes,
+        'edges': edges,
+    };
+}
+
+// Processing nodes for d3JS forcedSimulation's format
+function processNodes(nodes) {
+    let processedNodes = []
+    nodes.forEach((value) => {
+        processedNodes.push({ name: value.id, type: value.type });
+    });
+    return processedNodes;
+}
+
+// Processing edges for d3JS forcedSimulation's format
+function processEdges(nodes, links) {
+    let processedLinks = [];
+    for (let i = 0; i < links.length; ++i) {
+        let source = nodes.findIndex(j => j.name === links[i].source);
+        let target = nodes.findIndex(j => j.name === links[i].target);
+        processedLinks.push({ source, target });
+    }
+    return processedLinks;
+}
+
+// Specifying color of graph nodes
+function nodeColor(nodeType) {
+    switch (nodeType) {
+        case 'root':
+            return "#4D2205";
+        case 'project':
+            return "#0E4D05";
+        case 'read':
+            return "#052748";
+        case 'join':
+            return '#46054D'
+    }
+}
```

Comment:
I added a color here for the join node.  It might be slightly more readable to define a constant at the top of the file:

```
const COLORS = {
    'node-green': '#0E4D05',
    'node-blue': '#052748',
    ...
};
```

Then you can do:

```
case 'project':
    return COLORS['node-blue'];
```